src/app/features/add-network/use-add-network.tsx

```diff
@@ -39,16 +43,39 @@ const initialFormValues: AddNetworkFormValues = {
   bitcoinNetwork: 'mainnet',
 };
 
+function useInitialValues() {
+  const { state } = useLocation();
+
+  if (!state) {
+    return initialFormValues;
+  }
+
+  const network = state.network as NetworkConfiguration | undefined;
```

Comment:
Can we add Zod validation of the values coming from `useLocation`. Experience from send forms is that, without thorough checking of these values, we run into bugs.

We'll prevent these (or at least resolve much faster), if we strongly type, and runtime check, values passed around with route state.